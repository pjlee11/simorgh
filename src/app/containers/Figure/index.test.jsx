import React from 'react';
import FigureContainer from './index';
import { shouldShallowMatchSnapshot } from '../../helpers/tests/testHelpers';

const imageAlt =
  'Map of the UK displaying Syrian refugees and asylum seekers per 10000 population. Ranges from 0 to 17.';
const imageSrc =
  'https://ichef.bbci.co.uk/news/640/cpsprodpb/439A/production/_100960371_syrians_and_asylum_v2-nc.png';
const caption = 'This is a caption';
const copyrightText = 'Getty Images';

describe('Figure', () => {
  shouldShallowMatchSnapshot(
    'should render an image with alt text',
    <FigureContainer src={imageSrc} alt={imageAlt} />,
  );

  shouldShallowMatchSnapshot(
    'should render copyright text',
    <FigureContainer src={imageSrc} alt={imageAlt} copyright={copyrightText} />,
  );

  shouldShallowMatchSnapshot(
    'should render caption text',
    <FigureContainer src={imageSrc} alt={imageAlt} caption={caption} />,
  );
});
