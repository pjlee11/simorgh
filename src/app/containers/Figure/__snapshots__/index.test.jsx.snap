// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Figure should render an image with alt text 1`] = `
<Figure>
  <ImagePlaceholder>
    <Image
      alt="Map of the UK displaying Syrian refugees and asylum seekers per 10000 population. Ranges from 0 to 17."
      src="https://ichef.bbci.co.uk/news/640/cpsprodpb/439A/production/_100960371_syrians_and_asylum_v2-nc.png"
    />
  </ImagePlaceholder>
</Figure>
`;

exports[`Figure should render caption text 1`] = `
<Figure>
  <ImagePlaceholder>
    <Image
      alt="Map of the UK displaying Syrian refugees and asylum seekers per 10000 population. Ranges from 0 to 17."
      src="https://ichef.bbci.co.uk/news/640/cpsprodpb/439A/production/_100960371_syrians_and_asylum_v2-nc.png"
    />
  </ImagePlaceholder>
  <Text
    paragraphOverride={[Function]}
    text="This is a caption"
  />
</Figure>
`;

exports[`Figure should render copyright text 1`] = `
<Figure>
  <ImagePlaceholder>
    <Image
      alt="Map of the UK displaying Syrian refugees and asylum seekers per 10000 population. Ranges from 0 to 17."
      src="https://ichef.bbci.co.uk/news/640/cpsprodpb/439A/production/_100960371_syrians_and_asylum_v2-nc.png"
    />
    <Copyright>
      Getty Images
    </Copyright>
  </ImagePlaceholder>
</Figure>
`;
